File: src/main/java/net/arvandor/numinoustreasury/item/NuminousItemInventoryHolder.java
package net.arvandor.numinoustreasury.item;

import org.bukkit.Bukkit;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.InventoryHolder;
import org.bukkit.inventory.ItemStack;

public class NuminousItemInventoryHolder implements InventoryHolder {

    private final NuminousItemStack numinousItemStack;
    private final ItemStack itemStack;
    private final Inventory inventory;

    public NuminousItemInventoryHolder(NuminousItemStack numinousItemStack, ItemStack itemStack) {
        this.numinousItemStack = numinousItemStack;
        this.itemStack = itemStack;
        this.inventory = Bukkit.createInventory(this, numinousItemStack.getItemType().getInventorySlots(), numinousItemStack.getItemType().getName());
        ItemStack[] inventoryContents = numinousItemStack.getInventoryContents();
        if (inventoryContents != null) {
            this.inventory.setContents(inventoryContents);
        }
    }

    @Override
    public Inventory getInventory() {
        return inventory;
    }

    public void updateItemStack() {
        NuminousItemStack newNuminousItemStack = numinousItemStack.copy(
                null,
                null,
                inventory.getContents(),
                null
        );

        newNuminousItemStack.update(itemStack, false);
    }
}

File: src/main/kotlin/net/arvandor/numinoustreasury/item/NuminousItemInventoryHolder.kt
package net.arvandor.numinoustreasury.item

import org.bukkit.Bukkit
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.InventoryHolder
import org.bukkit.inventory.ItemStack

class NuminousItemInventoryHolder(private val numinousItemStack: NuminousItemStack, private val itemStack: ItemStack?) :
    InventoryHolder {
    private val inventory =
        Bukkit.createInventory(this, numinousItemStack.itemType.inventorySlots, numinousItemStack.itemType.name)

    init {
        val inventoryContents = numinousItemStack.inventoryContents
        if (inventoryContents != null) {
            inventory.contents = inventoryContents
        }
    }

    override fun getInventory(): Inventory {
        return inventory
    }

    fun updateItemStack() {
        val newNuminousItemStack =
            numinousItemStack.copy(
                null,
                null,
                inventory.contents,
                null,
            )

        newNuminousItemStack.update(itemStack, false)
    }
}
